[CI] MixedClusterClientYamlTestSuiteIT test {p0=indices.put_index_template/15_composition/Composable index templates that include subobjects: auto at root} failing

**Build Scans:**
- [elasticsearch-periodic #7811 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/b54m2hu5jgkzk)
- [elasticsearch-periodic #7805 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/xo6z3o2bx7yys)
- [elasticsearch-periodic #7799 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/mwj7tqrsdwjeg)
- [elasticsearch-periodic #7793 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/wg2ggnp2ttufg)
- [elasticsearch-periodic #7787 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/fn2pqhbjc2tgu)
- [elasticsearch-periodic #7781 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/qyqln7p4ir7bg)
- [elasticsearch-periodic #7775 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/ysik6tojjbuoq)
- [elasticsearch-periodic #7769 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/rbzsplmgxa2uq)
- [elasticsearch-periodic #7757 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/bp5wxlzfvtsky)
- [elasticsearch-periodic #7751 / 8.16.6_bwc](https://gradle-enterprise.elastic.co/s/u63ancrw72goy)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.16.6#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=index/92_metrics_auto_subobjects/Metrics object indexing}" -Dtests.seed=44E5483D8E6FCE36 -Dtests.bwc=true -Dtests.locale=sq-AL -Dtests.timezone=CTT -Druntime.java=24
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dindices.put_index_template%2F15_composition%2FComposable%20index%20templates%20that%20include%20subobjects%3A%20auto%20at%20root%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [index/92_metrics_auto_subobjects:8]: expected [2xx] status code but api [indices.put_index_template] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"composable template [test] template after composition is invalid","stack_trace":"org.elasticsearch.ElasticsearchException$1: composable template [test] template after composition is invalid\n\tat org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:707)\n\tat org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:635)\n\tat org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:201)\n\tat org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:160)\n\tat org.elasticsearch.server@8.19.0-SNAPSHOT/org.elasticsearch.
[truncated]
```

**Issue Reasons:**
- [8.x] 8 consecutive failures in step 8.16.6_bwc
- [8.x] 20 failures in test test {p0=indices.put_index_template/15_composition/Composable index templates that include subobjects: auto at root} (3.7% fail rate in 539 executions)
- [8.x] 20 failures in step 8.16.6_bwc (95.2% fail rate in 21 executions)
- [8.x] 20 failures in pipeline elasticsearch-periodic (95.2% fail rate in 21 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Looks like the same issue as https://github.com/elastic/elasticsearch/issues/127292.

```
[2025-05-30T18:25:26,726][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] [test] Stash dump on test failure [{
  "stash" : {
    "body" : {
      "error" : {
        "root_cause" : [
          {
            "type" : "illegal_argument_exception",
            "reason" : "composable template [test] template after composition is invalid",
            "stack_trace" : "org.elasticsearch.ElasticsearchException$1: composable template [test] template after composition is invalid
	at org.elasticsearch.server@8.16.6/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:706)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:634)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.rest.RestResponse.build(RestResponse.java:187)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:146)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:117)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:54)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:75)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:410)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:215)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:75)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:346)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:75)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:410)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:75)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:32)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:346)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:75)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:410)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:235)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:188)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:64)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:193)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$TemplateClusterStateUpdateTask.onFailure(MetadataIndexTemplateService.java:174)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$ExecutionResult.notifyFailure(MasterService.java:1009)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$ExecutionResult.onClusterStateUnchanged(MasterService.java:972)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:258)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.lang.IllegalArgumentException: composable template [test] template after composition is invalid
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:746)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:640)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:587)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:532)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:150)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:147)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245)
	... 11 more
Caused by: java.lang.IllegalArgumentException: invalid composite mappings for [test]
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1699)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:717)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1671)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:740)
	... 20 more
Caused by: org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:611)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:565)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:449)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1693)
	... 23 more
Caused by: org.elasticsearch.ElasticsearchParseException: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$Subobjects.from(ObjectMapper.java:86)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseSubobjects(ObjectMapper.java:344)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:609)
	... 26 more
"
          }
        ],
        "type" : "illegal_argument_exception",
        "reason" : "composable template [test] template after composition is invalid",
        "caused_by" : {
          "type" : "illegal_argument_exception",
          "reason" : "invalid composite mappings for [test]",
          "caused_by" : {
            "type" : "mapper_parsing_exception",
            "reason" : "Failed to parse mapping: unknown subobjects value: auto",
            "caused_by" : {
              "type" : "parse_exception",
              "reason" : "unknown subobjects value: auto",
              "stack_trace" : "org.elasticsearch.ElasticsearchParseException: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$Subobjects.from(ObjectMapper.java:86)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseSubobjects(ObjectMapper.java:344)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:609)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:565)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:449)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1693)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:717)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1671)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:740)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:640)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:587)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:532)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:150)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:147)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
"
            },
            "stack_trace" : "org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:611)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:565)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:449)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1693)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:717)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1671)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:740)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:640)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:587)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:532)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:150)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:147)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: org.elasticsearch.ElasticsearchParseException: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$Subobjects.from(ObjectMapper.java:86)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseSubobjects(ObjectMapper.java:344)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:609)
	... 26 more
"
          },
          "stack_trace" : "java.lang.IllegalArgumentException: invalid composite mappings for [test]
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1699)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:717)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1671)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:740)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:640)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:587)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:532)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:150)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:147)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:611)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:565)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:449)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1693)
	... 23 more
Caused by: org.elasticsearch.ElasticsearchParseException: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$Subobjects.from(ObjectMapper.java:86)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseSubobjects(ObjectMapper.java:344)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:609)
	... 26 more
"
        },
        "stack_trace" : "java.lang.IllegalArgumentException: composable template [test] template after composition is invalid
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:746)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:640)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:587)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:532)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:150)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:147)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1075)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1038)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:245)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1691)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1688)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1283)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.action.ActionListener.run(ActionListener.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1262)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.lang.IllegalArgumentException: invalid composite mappings for [test]
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1699)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:717)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateCompositeTemplate(MetadataIndexTemplateService.java:1671)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:740)
	... 20 more
Caused by: org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:611)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:565)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:449)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.lambda$validateCompositeTemplate$35(MetadataIndexTemplateService.java:1693)
	... 23 more
Caused by: org.elasticsearch.ElasticsearchParseException: unknown subobjects value: auto
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$Subobjects.from(ObjectMapper.java:86)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseSubobjects(ObjectMapper.java:344)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:452)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:122)
	at org.elasticsearch.server@8.16.6/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:609)
	... 26 more
"
      },
      "status" : 400
    }
  }
}]
```

--- a/qa/mixed-cluster/build.gradle
+++ b/qa/mixed-cluster/build.gradle
@@ -8,10 +8,13 @@
  */
 
 
-
 import org.elasticsearch.gradle.Version
 import org.elasticsearch.gradle.VersionProperties
 import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
+import org.elasticsearch.gradle.testclusters.TestClusterValueSource
+import org.elasticsearch.gradle.testclusters.TestClustersRegistry
+import org.elasticsearch.gradle.util.GradleUtils
+import org.elasticsearch.gradle.testclusters.TestClustersPlugin
 
 apply plugin: 'elasticsearch.internal-testclusters'
 apply plugin: 'elasticsearch.standalone-rest-test'

--- a/qa/mixed-cluster/build.gradle
+++ b/qa/mixed-cluster/build.gradle
@@ -8,13 +8,10 @@
  */
 
 
+
 import org.elasticsearch.gradle.Version
 import org.elasticsearch.gradle.VersionProperties
 import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
-import org.elasticsearch.gradle.testclusters.TestClusterValueSource
-import org.elasticsearch.gradle.testclusters.TestClustersRegistry
-import org.elasticsearch.gradle.util.GradleUtils
-import org.elasticsearch.gradle.testclusters.TestClustersPlugin
 
 apply plugin: 'elasticsearch.internal-testclusters'
 apply plugin: 'elasticsearch.standalone-rest-test'
@@ -89,6 +86,7 @@ buildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
         setting 'health.master_history.no_master_transitions_threshold', '10'
       }
       requiresFeature 'es.index_mode_feature_flag_registered', Version.fromString("8.0.0")
+      requiresFeature 'sub_objects_auto', Version.fromString("8.16.0")
       if (bwcVersion.before(Version.fromString("8.18.0"))) {
         jvmArgs '-da:org.elasticsearch.index.mapper.DocumentMapper'
         jvmArgs '-da:org.elasticsearch.index.mapper.MapperService'

